mono/mini/mini-generic-sharing.c

```diff
@@ -726,17 +726,21 @@ inflate_info (MonoRuntimeGenericContextInfoTemplate *oti, MonoGenericContext *co
 		MonoType *t = mono_class_inflate_generic_type_checked (m_class_get_byval_arg (dele_info->klass), context, error);
 		mono_error_assert_msg_ok (error, "Could not inflate generic type"); /* FIXME proper error handling */
 
-		MonoClass *klass = mono_class_from_mono_type_internal (t);
+		MonoClass *inflated_klass = mono_class_from_mono_type_internal (t);
 		mono_metadata_free_type (t);
 
 		MonoMethod *method = mono_class_inflate_generic_method_checked (dele_info->method, context, error);
 		mono_error_assert_msg_ok (error, "Could not inflate generic method"); /* FIXME proper error handling */
 
-		// FIXME: Temporary
-		MonoDelegateClassMethodPair *res = (MonoDelegateClassMethodPair *)mono_domain_alloc0 (domain, sizeof (MonoDelegateClassMethodPair));
+		MonoDelegateClassMethodPair* res = NULL;
```

Comment:
Does a similar change need to be made at https://github.com/Unity-Technologies/mono/blob/e492b1480b46367e556e2a092fe5c026492b16d9/mono/mini/mini-generic-sharing.c#L708?  Or this that not reproducible?

Reply:
In theory yes, but I didn't hit that code path at all. If we can figure out how to exercise that path I am fine fixing it as well.

There are also others that are only triggered when running with value type generic sharing (gsharedvt), which is not enabled in our fork (and never was).